server/meshmodel/policies/hierarchical-policy.rego

```diff
@@ -0,0 +1,46 @@
+
+# https://play.openpolicyagent.org/p/yLQ0On30CT
+package meshmodel_policy
+
+extract_components(services, selectors) = components {
+	components := [
+	component |
+		selector := selectors[_]
+		service := services[_]
+		selector.kind == service.type
+		component := service
+	]
+}
+
+has_key(x, k) {
+	_ = x[k]
+}
+
+evaluate_hierarchical_relationship(services, selectors) = service_name {
+
+	service_name := [service_name | 
```

Comment:
What does "service" refer to in this context? OAM leftovers? If so, can we stop propagating those?